File: C++/sliding_window/0121. Best Time to Buy and Sell Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxp=0;
        int l=0,r=0;
        while(r<prices.size()){
           if(prices[r]>prices[l]){
               maxp=max(maxp,prices[r]-prices[l]);
           }
           else{l=r;}
           ++r;
        }
        return maxp;
    }
};
//Time complexity : O(N) , space complexity:O(1)

File: javascript/sliding_window/0121. Best Time to Buy and Sell Stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let maxp=0
    let r=0, l=0
    while(r<prices.length){
        if(prices[r]>prices[l]){
            maxp=Math.max(maxp,prices[r]-prices[l])
        }
        else l=r;
        ++r;
    }
    return maxp
};